RandBLAS/dense.hh

```diff
@@ -68,33 +68,29 @@ struct DenseDist {
     const int64_t n_cols;
 };
 
-template <typename T>
+template <typename T, typename RNG = r123::Philox4x32>
 struct DenseSkOp {
-    const DenseDist dist;
-    const RNGState seed_state;
-    RNGState next_state;
-    const bool own_memory = true;
-    /////////////////////////////////////////////////////////////////////
-    //
-    //      Properties specific to dense sketching operators
-    //
-    /////////////////////////////////////////////////////////////////////
 
-    T *buff = nullptr;
-    bool filled = false;
-    bool persistent = true;
-    const blas::Layout layout = blas::Layout::ColMajor;
+    using generator = RNG;
+    using state_type = RNGState<RNG>;
+    using buffer_type = T;
+
+    const DenseDist dist;            ///< the name of the distribution and matrix size
+    const RNGState<RNG> seed_state;  ///< the initial CBRNG state
+    RNGState<RNG> next_state;        ///< the current CBRNG state
+    const bool own_memory = true;    ///< a flag that indicates who owns the memory
+
+    T *buff = nullptr;               ///< memory
+    bool filled = false;             ///< a flag that indicates if the memory was initialized
+    bool persistent = true;          ///< ???
+
+    const blas::Layout layout = blas::Layout::ColMajor; ///< matrix storage order
 
-    /////////////////////////////////////////////////////////////////////
-    //
-    //      Member functions must directly relate to memory management.
-    //
-    /////////////////////////////////////////////////////////////////////
```

Comment:
We should have prominent comments to differentiate between properties specific to dense sketching operators and properties that exist for any sketching operator. I would like the recurring design principles to be documented, if only briefly, wherever they appear.

Reply:
resolved in de486e0d